File: src/learning_tf_cpp/src/point_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/point_stamped.hpp"

using geometry_msgs::msg::PointStamped;
using namespace std::chrono_literals;

class PointPublisher : public rclcpp::Node {
public:
    PointPublisher() : Node("point_publisher_cpp"), x_(0.0) {
        RCLCPP_INFO(this->get_logger(), "坐标点发布方创建成功!");
        publisher_ = this->create_publisher<PointStamped>("point", 10);
        timer_ = this->create_wall_timer(0.1s, std::bind(&PointPublisher::timer_callback, this));
    }

private:
    void timer_callback() {
        auto msg = PointStamped();
        msg.header.stamp = this->now();
        msg.header.frame_id = "laser";
        msg.point.x = x_;
        msg.point.y = 0.0;
        msg.point.z = -0.5;
        x_ += 0.05;

        publisher_->publish(msg);
    }

    rclcpp::Publisher<PointStamped>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    double x_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<PointPublisher>());
    rclcpp::shutdown();

    return 0;
}

File: src/learning_tf_py/learning_tf_py/point_publisher.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import PointStamped

class PointPublisher(Node):
    def __init__(self):
        super().__init__("point_publisher_py")
        self.get_logger().info("坐标点发布方创建成功!(Python)")
        self.x_ = 0.0
        self.publisher_ = self.create_publisher(PointStamped, "point", 10)
        self.timer_ = self.create_timer(1.0, self.timer_callback)

    def timer_callback(self):
        msg = PointStamped()
        msg.header.stamp = self.get_clock().now().to_msg()
        msg.header.frame_id = "laser"
        msg.point.x = self.x_
        msg.point.y = 0.0
        msg.point.z = 0.3
        self.x_ += 0.05
        self.publisher_.publish(msg)

def main():
    rclpy.init()
    node = PointPublisher()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()

if __name__ == "__main__":
    main()